tenant/service_onboarding.go

```diff
@@ -98,7 +107,18 @@ func (s *OnboardService) onboardUser(ctx context.Context, req *influxdb.Onboardi
 
 	// create users password
 	if req.Password != "" {
-		s.service.SetPassword(ctx, user.ID, req.Password)
+		if err := s.service.SetPassword(ctx, user.ID, req.Password); err != nil {
```

Comment:
This looks like the best you can do in cascading errors.  Nice that you return the original error here, not the cleanup error. +1